faithcomesbyhearing/dbp
# FCBH-3322 Create new bulk content request end point

# Description

* Create showBatch to return bulk request
* Create blockedByBulkAccessControl to check fileset permissions and if. the current key allows the bulk request
* bulk endpoint created calls show functions without book and chapter to return all data within that fileset and its related bible

## Issue Link


[FCBH-3037](https://fullstacklabs.atlassian.net/browse/FCBH-3037)

## How Do I QA This
* Set env variable KEYS_FILESET_BULK_ACCESS={a key with filesets access in your local, or a group separated by commas}
* {basePath}/bibles/filesets/ENGESVP2DV/bulk – should return 404 if key does not have permissions, return a fileset if it does
* {basePath}/bibles/filesets/ENGESV/bulk - return all text for the whole bible fileset
* {basePath}/bibles/filesets/ENGESVN2DA/bulk - return mp3 files for each chapter of the bible for DA filesets
* {basePath}/bibles/filesets/ENGESVN2SA/bulk - return hls files for each chapter of the bible for SA filesets
* {basePath}/bibles/filesets/ENGESVP2DV/bulk – return mp4 files for each chapter of the bible for DV filesets

**Note:** To run the test suite refer to the Readme.md
